APE9.rst

```diff
@@ -0,0 +1,273 @@
+NDArr object will be .arr
+
+Need for ExposureInfo object?
+-----------------------------
+
+This is used by LSST to store information about the exposure such as filters,
+exposure time, etc. but all that is really needed is an isomorphic mapping
+to corresponding information in the astropy object. The question is: does
+it make sense to encapsulate that information in a single attribute, and 
+whether all think that these attributes are "astropy-universal”. For the 
+purposes of this, the only standard attribute is most likely the WCS.
+
+Items such as exposure time and filters could be relegated to the meta
+attribute. Making attributes such as these at the top level could be done
+through subclasses, though it may lead to many variants. Perhaps the best way
+to deal with this is for applications or libraries to state their requirements
+for items required to be in meta and leave it at that.
+
+NDData Slicing and Related
+++++++++++++++++++++++++++
+
+Slicing of NDData is supported (eventually) as well as striding. The corresponding 
+WCS attribute, if present, is updated accordingly. Eventually there may be support
+for slicing based on world coordinates, but this will not be part of the initial
+implementation.
+
+NDData does not support mathematical operations. Utilities will be provided to aid
+in how metadata items are to be propagated when the user merges NDData instances.
+
+Mask Object
++++++++++++
+
+The mask attribute now will be an instance of a Mask Object (but to support
+backward compatibilty, the attribute must have a different name than mask since
+the current implementation treats that attribute as a boolean array (or any
+object that mimics one). [An alternative option is to assume that the mask
+attribute of NDArr is a callable whereas the mask attribute of NDData represents
+a boolean array; this would be workable, but also likely confusing]
+This APE proposes that the new attribute have the name: lone_ranger. 
+Well, perhaps dmask.
+
+As objects, masks may take many internal forms. For example, they may be 
+simple boolean arrays, but they may also have many bitplanes or even be
+computed masks. The only requirement is that this object implement a
+callable interface such that it returns a boolean array when called with
+no positional or keyword arguments. These masks may implement any number
+of customizations to the callable (e.g., optional keyword arguments) as 
+well as methods to set the default behavior for the callable (i.e., when
+called with no arguments or keywords).
+The returned boolean array must be broadcastable to the underlying data array
+(a scalar boolean is permitted).
+
+As an example for bitplane masks:
+
+- ``nda.dmask()`` returns a boolean array
+
+- ``nda.dmask.apply_flags(…)`` sets how the bit planes are to be used to generate a boolean result when ``ndd.dmask()`` is used
+
+- ``nda.dmask(select=nda.dmask.bitplanes['saturated'] + nda.dmask.bitplanes['dead_pixels'])``
```

Comment:
Had you considered something like this, which might be more backwards compatible?
-  `nda.mask` a boolean-array-like object (numpy sub-class?)
- `nda.mask.select(nda.dmask.bitplanes['saturated'] + nda.dmask.bitplanes['dead_pixels'])`  does the same as your 3rd example
I think Perry asked previously for an example of how to implement this and it doesn't seem obvious without sub-classing numpy since we don't know how everyone is currently using mask.

Also, these labels are a good thing (along the lines of the meta-data abstraction I discussed previously) but can a bitmask also be allowed as an alternative? Using a single number is convenient, eg. for users, when dealing with some fixed convention.